pgdog/src/config/mod.rs

```diff
@@ -346,6 +350,40 @@ impl Config {
     pub fn multi_tenant(&self) -> &Option<MultiTenant> {
         &self.multi_tenant
     }
+
+    /// Get mirroring configuration for a specific source/destination pair.
+    pub fn get_mirroring_config(
```

Comment:
This does the search on every client connection. You would be better off storing this directly on the cluster, like we did with mirrors before and retrieving a pre-built list. You can think of `Cluster`, `Pool`, `Databases`, etc. as immutable configuration objects: generated from config on load time and store various settings, incl. mirroring. When we reload the config, old versions of them are destroyed and replaced with new ones with new config values.

This contrasts with `Client` which is long-lived and survives config reloads, that's why we fetch its config from the global config structure on each new query request.